src/startup_coordinator.c

```diff
@@ -293,42 +340,174 @@ static void StartupCoordinator_handle_start_time_proposal(StartupCoordinator *se
       LF_INFO(FED, "Start time negotiation completed Starting at " PRINTF_TIME, self->start_time_proposal);
       self->state = StartupCoordinationState_RUNNING;
       self->env->scheduler->set_and_schedule_start_tag(self->env->scheduler, self->start_time_proposal);
+      tag_t start_tag = {.time = self->start_time_proposal, .microstep = 0};
+      Environment_schedule_startups(self->env, start_tag);
+      Environment_schedule_timers(self->env, self->env->main, start_tag);
     } else {
       self->start_time_proposal_step++;
       send_start_time_proposal(self, self->start_time_proposal, self->start_time_proposal_step);
     }
   }
 }
 
+static void StartupCoordinator_handle_start_time_request(StartupCoordinator *self, StartupEvent *payload) {
+  FederatedEnvironment *env = (FederatedEnvironment *)self->env;
+  if (payload->neighbor_index == NEIGHBOR_INDEX_SELF) {
+    for (size_t i = 0; i < self->num_neighbours; i++) {
+      NetworkChannel *chan = env->net_bundles[i]->net_channel;
+      self->msg.which_message = FederateMessage_startup_coordination_tag;
+      self->msg.message.startup_coordination.which_message = StartupCoordination_start_time_request_tag;
+      lf_ret_t ret;
+      do {
+        ret = chan->send_blocking(chan, &self->msg);
+      } while (ret != LF_OK);
+
+      // We now schedule a system event here, because otherwise we will never detect no other federates responding
+      StartupCoordinator_schedule_system_self_event(self, self->env->get_physical_time(self->env) + MSEC(250),
+                                                    StartupCoordination_start_time_response_tag);
```

Comment:
Should MSEC(250) be a constant defined somewhere?

Reply:
I now solved it by having the `TRANSIENT_WAIT_TIME` macro which can be overwritten by the user, if he wants to wait a different amount.